Logic.py

```diff
@@ -79,20 +79,24 @@ def can_morph_ball(state: CollectionState, player: int) -> bool:
     return state.has(SuitUpgrade.Morph_Ball.value, player)
 
 
-def can_xray(state: CollectionState, player: int, hard_required: bool = False) -> bool:
+def can_xray(state: CollectionState, player: int, usually_required: bool = False, hard_required: bool = False) -> bool:

```

Comment:
What are your thoughts on having a single value rather than two booleans? Something like `RequirementType.USUALLY` and `RequirementType.ALWAYS`. I agree the `hard_required` could benefit from some more clear naming. I think having the enum would make it easier to read too:
`can_xray(state, player, RequirementType.ALWAYS)`

Reply:
I thought about that, but it would also mean I have to update every single call instead of just three XD